docs/DB_SYNCING.md

```diff
@@ -0,0 +1,54 @@
+# How to keep an external database in-sync with a Mesh node
+
+This guide will walk you through syncing an external database with a Mesh node so that the external database's state mirrors that of the Mesh node (and vice-versa). Whenever new orders are discovered or added to Mesh, they are inserted into the database. If an order is filled, cancelled, or has it's fillability changed it is updated or removed from the database. We are assuming that your database is storing both the order itself and how much of it is left to be filled (i.e., it's `fillableTakerAssetAmount`).
+
+## High-level architecture
+
+<img src="./mesh_db_sync_diagram.png" />
+
+Mesh is a stand-alone daemon that will be running independently from the rest of your infrastructure. Your backend will communicate with the Mesh node over [WebSockets](https://en.wikipedia.org/wiki/WebSocket) using [JSON-RPC](https://www.jsonrpc.org/). This is a bi-directional connection allowing both your server to notify Mesh of new orders, and Mesh to notify your server of order updates.
+
+## Initial sync
+
+When first connecting the DB and Mesh node, we first need to make sure both have the same orders and order-relevant state stored. We do this with the following steps:
+
+#### 1. Subscribe to Mesh
+
+Subscribe to the Mesh node's `orders` subscription over a WS connection. This can be done using our [golang](https://godoc.org/github.com/0xProject/0x-mesh/rpc) or [Typescript/Javascript](./json-rpc-clients/typescript) clients or any other JSON-RPC WebSocket client. Whenever you receive an order event from this subscription, make the appropriate updates to your DB. Each order event has an associated [OrderEventKind](https://godoc.org/github.com/0xProject/0x-mesh/zeroex#pkg-constants).
+
+| Kind                                       | DB operation                    |
+|--------------------------------------------|---------------------------------|
+| ADDED                                      | Insert                    |
+| FILLED                                     | Update |
+| FULLY_FILLED, EXPIRED, CANCELLED, UNFUNDED | Remove                    |
+| FILLABILITY_INCREASED                      | Upsert             |
+
+**Note:** Updates refer to updating the order's `fillableTakerAssetAmount` in the DB.
+
+#### 2. Get all orders currently stored in Mesh
+
+There might have been orders stored in Mesh that the DB doesn't know about at this time. Because of this, we must fetch all currently stored order in the Mesh node and upsert them in the database. This can be done using the [mesh_getOrders](./USAGE.md#mesh_getorders) JSON-RPC method. This method creates a snapshot of the Mesh node's internal DB of orders when first called, and allows for subsequent paginated requests against this snapshot. Because we are already subscribed to order events, any new orders added/removed after the snapshot is made will be discovered via that subscription.
```

Comment:
```suggestion
There might have been orders stored in Mesh that the DB doesn't know about at this time. Because of this, we must fetch all currently stored orders in the Mesh node and upsert them in the database. This can be done using the [mesh_getOrders](./USAGE.md#mesh_getorders) JSON-RPC method. This method creates a snapshot of the Mesh node's internal DB of orders when first called, and allows for subsequent paginated requests against this snapshot. Because we are already subscribed to order events, any new orders added/removed after the snapshot is made will be discovered via that subscription.
```